src/x.c

```diff
@@ -251,6 +251,34 @@ void x_move_win(Con *src, Con *dest) {
     }
 }
 
+/*
+ * Reparents child window from old con into new con. The reparenting happens in
+ * the next call of x_push_changes(). Unlike x_reparent_child & x_move_win, this
+ * function assumes that the old con is about to be destroyed and preserves
+ * window_rect and child_mapped.
+ *
+ */
+void x_reparent_child2(Con *new, Con *old) {
```

Comment:
Maybe a name like this would be a bit better?

```suggestion
void x_reparent_child_deep(Con *new, Con *old) {
```

Reply:
Agreed